src/main/kotlin/com/intershop/gradle/isml/IsmlPlugin.kt

```diff
@@ -65,7 +68,7 @@ open class IsmlPlugin : Plugin<Project> {
 
             if (extension.sourceSets.findByName(IsmlExtension.ISML_MAIN_SOURCESET) == null) {
                 val mainIsmlSourceSet = extension.sourceSets.create(IsmlExtension.ISML_MAIN_SOURCESET)
-                mainIsmlSourceSet.srcDir.set(layout.projectDirectory.dir(IsmlExtension.MAIN_TEMPLATE_PATH))
+                mainIsmlSourceSet.srcDir.set(layout.projectDirectory.dir(IsmlExtension.MAIN_TEMPLATE_PATH + "/" + project.name))
```

Comment:
is the project.name in the path really necessary. it looks like doubled:
rootporoject/subprojecgroup/**realsubproject**/src/main/isml/**realsubproject**